ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -165,23 +165,37 @@ def patch():
     if not in_aws_lambda() and not _has_patch_module():
         return
 
-    handler, handler_module, wrapper = _get_handler_and_module()
+    try:
+        handler, handler_module, wrapper = _get_handler_and_module()
 
-    if getattr(handler_module, "_datadog_patch", False):
-        return
-    setattr(handler_module, "_datadog_patch", True)
+        if getattr(handler_module, "_datadog_patch", False):
+            return
+        setattr(handler_module, "_datadog_patch", True)
```

Comment:
Should we set the attr after wrap()?

Reply:
I think the order here doesn't matter, but what do you think?

Reply:
if the `wrap()` fail, it will stil set attr `_datadog_path` to True, right?

Reply:
You're right, i'll switch the spots of the unpatch too.